power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -364,16 +603,16 @@ class Deserializer {
             // So we disable the good code and write the loop manually
             // std::transform_inclusive_scan(msg_data.cbegin(), msg_data.cend(),
             //                               buffer.indptr.begin() + 1,
-            //                               std::plus{}, [](auto const& x) { return static_cast<Idx>(x.size()); });
```

Comment:
can you try to check whether this still is an issue since we now bumped to newer apple clang? if so, we can remove these comments

Reply:
Still fails. After some digging into [cpp reference](https://en.cppreference.com/w/cpp/algorithm/transform_inclusive_scan), we might have done something wrong before.

We did not provide `T init` in the call. It will be inferred by the `decltype(first)::value_type)`, which should not compile. So apple clang is actually giving the correct error.

Reply:
Let's see after adding `init` if everything can compile.

Reply:
it compiles!